AppSettingsManagerAPI/AppSettingsManagerApi.Infrastructure/MySql/MySqlSettingRepository.cs

```diff
@@ -23,14 +24,16 @@ public class MySqlSettingRepository : ISettingRepository
         _settingsConverter = settingsConverter;
     }
 
-    public async Task<Model.Setting> GetSetting(string settingId, int version)
+    public async Task<Model.Setting> GetSetting(string settingId, int version, string userId, string password)
     {
         // Call .Single() because there should only be one entry with this id/version
         // and .Single() will return a single object rather than a list
         var setting = await _settingsContext.Settings.SingleAsync(
             s => s.SettingGroupId == settingId && s.Version == version
         );
 
+        var permissions = await setting.Permissions.Single(permission => permission.UserId == UserId && permission.Password == Password); //Permissions
+
         return _settingsConverter.Convert(setting);
     }
```

Comment:
```suggestion
        var permission = await setting.Permissions.Single(permission => permission.UserId == UserId && permission.Password == Password); //Permissions

        return permission.CurrentPermissionLevel != PermissionLevel.None ? _settingsConverter.Convert(setting) : throw new Exception("Insufficient permission");
    }
```

This is good, it'll throw an exception if the UserId/Password isn't found. I forgot to mention we should still check the type of permission that the user has since we have a None type. Could do something like this with a ternary operator